--- a/libstdc++-v3/config/os/gnu-linux/os_defines.h
+++ b/libstdc++-v3/config/os/gnu-linux/os_defines.h
@@ -38,15 +38,8 @@
 
 #include <features.h>
 
-// Provide a declaration for the possibly deprecated gets function, as
-// glibc 2.15 and later does not declare gets for ISO C11 when
-// __GNU_SOURCE is defined.
-#if __GLIBC_PREREQ(2,15) && defined(_GNU_SOURCE)
-# undef _GLIBCXX_HAVE_GETS
-#endif
-
 // Glibc 2.23 removed the obsolete isinf and isnan declarations. Check the
 // version dynamically in case it has changed since libstdc++ was configured.
-#define _GLIBCXX_NO_OBSOLETE_ISINF_ISNAN_DYNAMIC __GLIBC_PREREQ(2,23)
+#define _GLIBCXX_NO_OBSOLETE_ISINF_ISNAN_DYNAMIC 1
 
 #endif
